File: 383-RansomNote/383-RansomNote.c
// Last updated: 08/04/2026, 12:39:53
bool canConstruct(char* ransomNote, char* magazine) {

    int r_length = strlen(ransomNote);
    int m_length = strlen(magazine);
    bool char_exist;

    for (int i = 0; i < r_length; i++) {
        char_exist = false;
        for (int j = 0; j < m_length; j++) {
            if (ransomNote[i] == magazine[j]) {
                magazine[j] = '0';
                char_exist = true;
                break;
            } 
        }

        if (char_exist == false) {
            return false;
        }
    }
    return true;
}

File: 383-RansomNote/383-RansomNote.py
# Last updated: 18/08/2026, 14:56:17
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        magazineFreq = {}
        ransomFreq = {}

        for i in range(len(magazine)):
            magazineFreq[magazine[i]] = magazineFreq.get(magazine[i], 0) + 1
        
        for i in range(len(ransomNote)):
            ransomFreq[ransomNote[i]] = ransomFreq.get(ransomNote[i], 0) + 1
            if ransomNote[i] not in magazineFreq or ransomFreq[ransomNote[i]] > magazineFreq[ransomNote[i]]:
                return False
        
        return True
